notebooks/MAST/HSC/HSC_TAP/HSC_TAP.ipynb

```diff
@@ -0,0 +1,472 @@
+    "\n",
+    "For another example of using data from a TAP service, we start by doing a search around the SMC with a .25 degree radius for objects with ACS F555W and F814W measurements. HSC TAP will limit us to 100k responses by default. Note this is a large query that can take over a minute to run. See <a href=\"http://archive.stsci.edu/hst/hsc/help/use_case_2_v3.html\" target=\"_blank\">HSC Use Case 2</a> for more details."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "t0 = time.time()\n",
+    "\n",
+    "job = HSC_service.launch_job_async(\"\"\"\n",
+    "SELECT MatchID, MatchRA, MatchDec, CI, A_F555W, A_F814W\n",
+    "FROM dbo.SumMagAper2CatView\n",
+    "WHERE A_F555W_N > 0 and A_F814W_N > 0\n",
+    "    AND CONTAINS(POINT('ICRS', MatchRA, MatchDec),CIRCLE('ICRS',13.1866,-72.8286,0.25))=1\n",
+    "   \"\"\")\n",
+    "HSC_results = job.get_results()\n",
+    "print(\"Query completed in {:.1f} sec\".format(time.time()-t0))\n",
+    "HSC_results.pprint()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, plot the color-magnitude diagram for the ~100k points retrieved from the database. This uses kernel density estimate for the crowded plot, which can take a few minutes to complete."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "f555w = HSC_results['A_F555W']\n",
+    "f814w = HSC_results['A_F814W']\n",
+    "VminusI = f555w-f814w\n",
+    "CI = HSC_results['CI']\n",
+    "w = numpy.where((CI>0.9) & (CI<1.6) & (VminusI > -1.5) & (VminusI < 1.5))\n",
+    "print(len(w[0]),\"points remaining after CI and V-I filtering\")\n",
+    "\n",
+    "# Calculate the point density\n",
+    "x = numpy.array(VminusI[w])\n",
+    "y = numpy.array(f555w[w])\n",
+    "xy = numpy.vstack([x,y])\n",
+    "\n",
+    "t0 = time.time()\n",
+    "z = gaussian_kde(xy)(xy)\n",
+    "print(\"kde took {:.1f} sec\".format(time.time()-t0))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# Sort the points by density, so that the densest points are plotted last\n",
+    "idx = z.argsort()\n",
+    "x, y, z = x[idx], y[idx], z[idx]\n",
+    "\n",
+    "pylab.rcParams.update({'font.size': 16})\n",
+    "pylab.figure(1,(12,10))\n",
+    "pylab.scatter(x, y, c=z, s=2, edgecolor='', cmap='plasma')\n",
+    "pylab.autoscale(tight=True)\n",
+    "pylab.xlabel('V-I')\n",
+    "pylab.ylabel('V')\n",
+    "pylab.gca().invert_yaxis()\n",
+    "pylab.colorbar()\n",
+    "pylab.text(.17,.93,'{:d} stars in SMC'.format(len(x)),\n",
+    "       horizontalalignment='center',\n",
+    "       transform=pylab.gca().transAxes)\n",
+    "pylab.savefig(\"smc_colormag.png\")"
```

Comment:
why the savefig?  That seems unnecessary since it's already being displayed in the notebook.

Reply:
Commented out.